src/Components/Address/AddressForm.tsx

```diff
@@ -107,7 +102,13 @@ export const AddressForm: React.FC<AddressFormProps> = ({
   const changeEventHandler = (key: keyof Address) => (
     ev: React.FormEvent<HTMLInputElement>
   ) => {
-    const shouldFetch = isAddressAutocompleteEnabled && key === "addressLine1"
+    const inputValue = ev.currentTarget.value
+
+    // Only trigger fetch for autocomplete if the key is 'addressLine1' and the input length is >= 3
+    const shouldFetch =
+      isAddressAutocompleteEnabled &&
+      key === "addressLine1" &&
+      inputValue.length >= 3
```

Comment:
Currently, the check for input length is [inside the autocomplete hook](https://github.com/artsy/force/blob/3e86fa5931dffe034d3d7b2dd67cd06a2aaeb414/src/Components/Address/useAddressAutocomplete.ts#L83-L86). Can you put this check there?

Reply:
Yup, I kinda messed up and placed it in the wrong location.